htdocs/compta/cashcontrol/report.php

```diff
@@ -135,6 +135,8 @@
 	dol_print_error('Value for key = '.$key.' not supported');
 	exit;
 }*/
+if ($object->date_valid==null) $object->date_valid=dol_now();
+$sql .= " AND f.date_closing BETWEEN '".$db->idate($object->date_creation)."' AND '".$db->idate($object->date_valid)."'";
 if ($syear && !$smonth) {
 	$sql .= " AND datef BETWEEN '".$db->idate(dol_get_first_day($syear, 1))."' AND '".$db->idate(dol_get_last_day($syear, 12))."'";
```

Comment:
Same question, may be we should include the date of payment here instead of the date of invoice ? What do you think ?